lmnet/lmnet/datasets/bdd100k.py

```diff
@@ -74,41 +82,88 @@ def __init__(self,
         self.num_workers = num_workers
 
         subset_dir = "train" if subset == "train" else "val"
-        self.img_dir = join(self.data_dir, "images", "100k", subset_dir)
-        self.anno_dir = join(self.data_dir, "labels", "100k", subset_dir)
+        self.img_dir = os.path.join(self.data_dir, "images", "100k", subset_dir)
+        self.anno_dir = os.path.join(self.data_dir, "labels", "bdd100k_labels_images_" + subset_dir + ".json")
         self.paths = []
         self.bboxs = []
+        self.save_file = os.path.join("dataset", "bdd100k_%s.pickle" % self.subset)
+        if not os.path.exists(os.path.dirname(self.save_file)):
+            os.makedirs(os.path.dirname(self.save_file))
 
-        self._init_files_and_annotations()
+        if os.path.exists(self.save_file):
+            with open(self.save_file, "rb") as f:
+                loaded_data = pickle.load(f)
+                if len(loaded_data[0]) > 0:
+                    self.paths, self.bboxs = loaded_data
+                else:
+                    os.remove(self.save_file)
+                    self._init_files_and_annotations()
+        else:
+            self._init_files_and_annotations()
 
```

Comment:
Please don't save the cache file.

Reply:
The reason I saved the cached data as `pickle` is for convenience. Every time you start the training with BDD100K the annotation preparation takes about 10min (on server5). 

Anyway, If the caching is not recommended, I can remove the pickle saving and every time training run it will take around 10min (on server 5) to prepare the lists.

Reply:
I wrapped whole lines of `_init_files_and_annotations` by `time.time()`, measured the performance. Here's my result on  the server 5.

```
elapsed time for  70000  images: 8.224702596664429 sec
elapsed time for  10000  images: 2.0470070838928223 sec
```

This performance seems that the data is already on the disk cache, hence it would be the best case. However, even if the data is not on the disk cache, since server 5's storage is SSD, the performance would not be changed so drastically. 10 mins vs 10 secs are very big difference, I want to know what causes this difference. Could you tell me how to reproduce your result?

Reply:
@tkng San:
The timing looks like it is still reading from the pickle file. Did you remove the pickle files?
Please delete the pickle files:
```
# from blueoil/lmnet
$ rm -rf dataset/bdd100k_*.pickle
```
Anyway, I've modified to use dict instead of list as suggested by Hanamoto San. Now without parallel implementation the dataset prepared within 1min (in server 5)

Reply:
@tkng Only mention one detail that maybe helps. 
On our servers we usually have the next configuration:
```
/  is mounted on SSD
/storage  is mounted on a traditional HDD
/storage2  is mounted on a traditional HDD
```
Normally, we store the datasets on `/storage` or `/storage2` so we are always reading from HDD drives during training. Always wondered if performance will improve if we store the dataset on the SSD. Or we actually keep most of the dataset in memory during training?

Not sure how it works, actually.

Reply:
This problem is already solved. 
Please see this comment : https://github.com/blue-oil/blueoil/pull/399#discussion_r318364484